storagenode/trust/service.go

```diff
@@ -131,7 +132,7 @@ func (pool *Pool) GetSignee(ctx context.Context, id storj.NodeID) (_ signing.Sig
 	defer info.mu.Unlock()
 
 	if info.identity == nil {
-		identity, err := pool.kademlia.FetchPeerIdentity(ctx, id)
+		identity, err := pool.dialer.FetchPeerIdentityUnverified(ctx, address.Address)
```

Comment:
How do we verify that the uplink has sent us the correct address and signature?

Reply:
whoops, i should have compared identity.ID to the `id` variable. if they don't match, then we failed to get the public key for the id we actually cared about. good catch